compiler/resolution/resolveFunction.cpp

```diff
@@ -1540,7 +1540,7 @@ void resolveIfExprType(CondStmt* stmt) {
       CallExpr* call = toCallExpr(refBranch->body.tail);
       SymExpr* rhs = toSymExpr(call->get(2));
       if (typeNeedsCopyInitDeinit(rhs->getValType())) {
-        CallExpr* copy = new CallExpr(astr_autoCopy, rhs->remove());
+        CallExpr* copy = new CallExpr(astr_autoCopy, rhs->remove(), gFalse);
```

Comment:
Shouldn't this use isConst based on ret having FLAG_CONST? Try that and add a test of an if-expr making a const domain.

Reply:
You are right and wrong :)

I have changed the implementation as you suggest as it makes sense in general. However, at least in the simple cases, copy elision removes the copy added here. So, I also changed the implementation in the build.cpp to look at `chpl__buildDomainExpr` within potentially nested if expressions.